framework/helpers/BaseStringHelper.php

```diff
@@ -457,17 +457,23 @@ public static function mb_ucfirst($string, $encoding = 'UTF-8')
      * @param string $string the string to be proceeded
      * @param string $encoding Optional, defaults to "UTF-8"
      * @return string
-     * @see https://secure.php.net/manual/en/function.ucwords.php
+     * @see https://www.php.net/manual/en/function.ucwords
      * @since 2.0.16
      */
     public static function mb_ucwords($string, $encoding = 'UTF-8')
     {
-        $words = preg_split("/\s/u", $string, -1, PREG_SPLIT_NO_EMPTY);
+        if (empty($string)) {
+            return '';
+        }
 
-        $titelized = array_map(function ($word) use ($encoding) {
-            return static::mb_ucfirst($word, $encoding);
-        }, $words);
+        $parts = preg_split('/(\s+[^\w]+\s+|\s+)/u', (string) $string, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
+        $upEven = !trim($parts[0]);
+        foreach ($parts as $key => $value) {
+            if ($upEven !== !($key & 1)) {
```

Comment:
What's the logic behing it?

Reply:
`$string` converts to array: `[word, space, word, space, ...]` or `[space, word, space, word, space, ...]`,  method converts odd/event values base on check: is first symbol of first value is whitespace?

Reply:
So `!($key & 1)` returns what? Is it possible to make more understandable?

Reply:
@samdark see [first example](https://www.php.net/manual/ru/function.array-filter.php) `($key % 2) === 0` is "more readable"? (:

Reply:
Yes, it is more readable. And change the name of that variable to something more meaningful.

Reply:
@bizley
> And change the name of that variable

I thought about it, but I couldn't think something better.

Reply:
A bit, yes.